specs/deneb/p2p-interface.md

```diff
@@ -65,34 +91,32 @@ The new topics along with the type of the `data` field of a gossipsub message ar
+Deneb introduces new global topics for blob sidecars.
 
 ##### `beacon_block`
 
-This topic is deprecated and clients **MUST NOT** expose in their topic set to any peer. Implementers do not need to do
-anything beyond simply skip implementation, and it is explicitly called out as it is a departure from previous versioning
-of this topic.
+The *type* of the payload of this topic changes to the (modified) `SignedBeaconBlock` found in deneb.
 
-Refer to [the section below](#transitioning-the-gossip) for details on how to transition the gossip.
+##### `blob_sidecar_{index}`
 
-##### `beacon_block_and_blobs_sidecar`
+This topic is used to propagate signed blob sidecars, one for each sidecar index. The number of indices is defined by `MAX_BLOBS_PER_BLOCK`.
 
-This topic is used to propagate new signed and coupled beacon blocks and blobs sidecars to all nodes on the networks.
+The following validations MUST pass before forwarding the `sidecar` on the network, assuming the alias `sidecar = signed_blob_sidecar.message`:
 
-In addition to the gossip validations for the `beacon_block` topic from prior specifications, the following validations MUST pass before forwarding the `signed_beacon_block_and_blobs_sidecar` on the network.
-Alias `signed_beacon_block = signed_beacon_block_and_blobs_sidecar.beacon_block`, `block = signed_beacon_block.message`, `execution_payload = block.body.execution_payload`.
-- _[REJECT]_ The KZG commitments correspond to the versioned hashes in the transactions list
-  -- i.e. `verify_kzg_commitments_against_transactions(block.body.execution_payload.transactions, block.body.blob_kzg_commitments)`
+- _[REJECT]_ The sidecar is for the correct topic -- i.e. `sidecar.index` matches the topic `{index}`.
+- _[IGNORE]_ The sidecar is not from a future slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance) -- i.e. validate that `sidecar.slot <= current_slot` (a client MAY queue future blocks for processing at the appropriate slot).
+- _[IGNORE]_ The sidecar is from a slot greater than the latest finalized slot -- i.e. validate that `sidecar.slot > compute_start_slot_at_epoch(state.finalized_checkpoint.epoch)`
+- _[IGNORE]_ The blob's block's parent (defined by `sidecar.block_parent_root`) has been seen (via both gossip and non-gossip sources) (a client MAY queue blocks for processing once the parent block is retrieved).
+- _[REJECT]_ The proposer signature, `signed_blob_sidecar.signature`, is valid with respect to the `sidecar.proposer_index` pubkey.
+- _[IGNORE]_ The sidecar is the only sidecar with valid signature received for the tuple `(sidecar.slot, sidecar.proposer_index, sidecar.index)`.
+  -- If full verification of the blob fails at a later processing stage, clients MUST clear the blob from this "seen" cache so as to allow a the valid blob to propagate. The next block producer MAY orphan the block if they have observed multiple blobs signed by the proposer for the same "seen" tuple.
```

Comment:
```suggestion
  -- If full verification of the blob fails at a later processing stage, clients MUST clear the blob from this "seen" cache so as to allow a potential valid blob to propagate.
  -- The next block producer MAY orphan the block if they have observed multiple blobs signed by the proposer for the same "seen" tuple.
```

Reply:
Do we need to put this note about clearing the seen cache? We have the same problem with attestations and blocks today. IIUC this cache and IGNORE condition can slow down propagation, but IWANT/IHAVE and subsequently querying for message dependencies will still allow getting such messages even if the cache here hits.

Curious your perspective on this before merge

Reply:
> We have the same problem with attestations and blocks today.

In the past, we've considered it "risky" enough for the proposer to try to poison this seen cache because slashing conditions.

Now that I think about it, maybe the most simple solution, rather than playing these "first" games, is to remove the condition altogether - we *want* duplicates to propagate to a certain extent so they reach the validators that are in a position to enforce the rules.

I've opened https://github.com/ethereum/consensus-specs/pull/3257 which covers blocks and attestations. 

The rule here is a bit more nuanced in that there's no actual slashing message to rely on - instead, we would rely on the "softer" enforcement of having the next honest proposer orphan such a history. Having written #3257, I think the best course of action is actually to remove the rule altogether and allow implementations to do spam prevention measures individually, for example by limiting to N observed tuples or IGNORE:ing duplicates outside of a much tighter 4s window.

Reply:
> but IWANT/IHAVE and subsequently querying for message dependencies 

IGNORE messages are also not given to the IWANT message cache, they're gone.

Reply:
I still don't think we should clear the seen cache. In the event of a malicious proposer, message dependencies are the best path forward, imo. Gossip will be much slower for such a dishonest proposer, potentially leading to orphaning, but due to dependencies not a permanent split

note, getting a block from a peer doesn't necessarily mean they have the *correct* blobs, but getting an attestation or child block from a peer does. So these dependency messages become signals on who to download the blobs from and won't lead to unresolvable partitions.

Additionally, if you don't get any dependency messages, then the block will be orphaned. which was at the direct  cause of the proposer trying to dos the blobs topic.

So I'd say, keep the cache.

Reply:
My preference is to keep the cache as we do in other message types and to take this to an issue for discussion. Instead of blocking on this to merge the bulk of the PR

Reply:
54d2559eb54db217844d30098662e0be3c8d9e5c - good point about the block *likely* being orphaned anyway - let's hope that is discouragement enough.